src/backend/pipeline/cont_scheduler.c

```diff
@@ -160,18 +165,18 @@ GetContQueryProcName(ContQueryProc *proc)
 
 	switch (proc->type)
 	{
-	case Combiner:
-		sprintf(buf, "%s%d [%s]", "combiner", proc->group_id, NameStr(proc->group->db_name));
-		break;
-	case Worker:
-		sprintf(buf, "%s%d [%s]", "worker", proc->group_id, NameStr(proc->group->db_name));
-		break;
-	case Scheduler:
-		return pstrdup("scheduler");
-		break;
-	case Adhoc:
-		return pstrdup("adhoc");
-		break;
+		case Combiner:
+			sprintf(buf, "combiner%d [%s]", proc->group_id, NameStr(proc->db_meta->db_name));
+			break;
+		case Worker:
+			sprintf(buf, "worker%d [%s]", proc->group_id, NameStr(proc->db_meta->db_name));
+			break;
+		case Adhoc:
+			sprintf(buf, "adhoc [%s]", NameStr(proc->db_meta->db_name));
+			break;
+		case Scheduler:
+			return "scheduler";
```

Comment:
Callers should expect to `pfree` whatever this function returns, so this should be `pstrduped`.